### PR TITLE
Update integration tests for new DynamicHMC versions

### PR DESCRIPTION
**Pull request by burtonjosh**
Noticed that some of the integration tests weren't passing, looks like at some point the `chain` field has been renamed to `posterior_matrix` in AdvancedHMC. Also the `mean_and_mcse` function was not working "per parameter" so I made some changes to fix that.

### CODE REVIEW

**Review by sethaxen — APPROVED** [review]
Thanks @burtonjosh for the PR! LGTM, just one tiny ~whitespace~ change.

Relates #90

File: test/integration/DynamicHMC/runtests.jl [hunk 1]
```diff
@@ -36,13 +36,13 @@ end
 
 function mean_and_mcse(f, θs)
     zs = map(f, θs)
-    ms = mean(zs)
-    ses = map(mcse, eachrow(reduce(hcat, zs)))
+    ms = mean(zs,dims=2)
```
[the remark this answers is not in the document]
Comment (sethaxen) [reply test/integration/DynamicHMC/runtests.jl:39]: @burtonjosh did you forget to push the fix?

[the remark this answers is not in the document]
Comment (burtonjosh) [reply test/integration/DynamicHMC/runtests.jl:39]: sorry I thought I had approved this!

[the remark this answers is not in the document]
Comment (burtonjosh) [reply test/integration/DynamicHMC/runtests.jl]: Oh my bad, I just resolved it without first pushing the fix - sorted now

[the remark this answers is not in the document]
Comment (sethaxen) [reply test/integration/DynamicHMC/runtests.jl]: No problem, thanks!

**Pull request merged**